client/src/components/HomePage/Registration.jsx

```diff
@@ -45,10 +80,12 @@ class Registration extends Component {
 
         if ('error' in data) {
           this.setState({ [`check_${data.error}`]: `Invalid ${data.error}` })
-          this.props.history.push('/registration');
           return null;
+
         } else {
-          this.props.history.push('/');
+        window.Materialize.toast("Check your email box", 3000);
+        window.history.forward('/registration-info');
```

Comment:
Wrong indentation. Also, you don't need this `return null`, you don't have any code after this `if`.